framework/db/command_register.py

```diff
@@ -51,5 +51,5 @@ def CommandAlreadyRegistered(self, original_command, target_id=None):
                 return self.target.GetTargetURLForID(register_entry.target_id)
             else:  # Either command failed or plugin output doesn't exist
                 self.DeleteCommand(original_command)
-                return self.target.GetTargetURLForID(register_entry.target_id)
```

Comment:
@delta24 the reason it is not running first time is because when the force overwrite is given it deletes the plugin output and when it comes to this function since command is registered but plugin output doesn't exist the `else` statement is executed which doesn't return `None` due to which `CanRunCommand()` returns false and command is not executed. But in next time since command doesn't run the previous time this function return `None` and  `CanRunCommand()` returns `True` and command is executed. So we have to change something here but now I think this may not be a good solution , I will look for a better solution or do you have something in mind @delta24